### PR TITLE
 (BOLT-372) Add plan commands

### PR DESCRIPTION
**Pull request by adreyer**

### CODE REVIEW

File: lib/orchestrator_client.rb [hunk 1]
```diff
@@ -55,7 +55,7 @@ def post(location, body)
     req.body = body.to_json
     res = https.request(req)
 
-    if res.code != "202"
+    unless Set.new(["202", "200"]).include? res.code
```
Comment (MikaelSmith) [inline lib/orchestrator_client.rb]: Using a one-time set here brings no actual benefit.

File: lib/orchestrator_client/command.rb [hunk 2]
```diff
@@ -4,18 +4,33 @@ def initialize(https)
...
 
-  def task(options = {})
+  def deploy(options = {})
     raise ArgumentError, 'Must pass options as a hash' unless options.is_a? Hash
-    @https.post("command/task", options)
+    @https.post("command/deploy", options)
   end
 
-  def deploy(options = {})
+  def plan_start(options = {})
```
Comment (MikaelSmith) [inline lib/orchestrator_client/command.rb]: When will plan_start/plan_finish be used? Or how are they exposed?

File: lib/orchestrator_client.rb (hunk 1 above)
Comment (adreyer) [reply lib/orchestrator_client.rb]: what is the other ruby way to do this?

File: lib/orchestrator_client/command.rb (hunk 2 above)
Comment (adreyer) [reply lib/orchestrator_client/command.rb]: `client.command.plan_start`

File: lib/orchestrator_client.rb (hunk 1 above)
Comment (adreyer) [reply lib/orchestrator_client.rb]: oh for some reason I thought arrays responded to include based on their indexes

Comment (MikaelSmith) [reply lib/orchestrator_client.rb]: https://ruby-doc.org/core-2.2.0/Array.html#method-i-include-3F

File: lib/orchestrator_client/command.rb (hunk 2 above)
Comment (MikaelSmith) [reply lib/orchestrator_client/command.rb]: 🤷‍♂️

Comment (adreyer) [reply lib/orchestrator_client/command.rb]: bolt will call it directly in other words. The orchestrator transport is informed that a plan is running before the plan starts. It then calls start plan on the client before running any tasks. `finish_task` is called after the plan finishes and orchestrator calls finish_task on any clients that were used during the plan.

**Pull request merged**